portfolio/src/main/java/com/google/sps/servlets/NicknameServlet.java

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+
+import com.google.gson.Gson;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.users.User;
+
+/** Servlet that allows for adding of nicknames. */
+@WebServlet("/nickname")
+public class NicknameServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    User user = userService.getCurrentUser();
+    if (user == null) {
+        response.sendRedirect("/index.html");
+        return;
+    }
+
+    String id = user.getUserId();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("User")
+        .setFilter(new Query.FilterPredicate("id", Query.FilterOperator.EQUAL, id));
+
+    PreparedQuery results = datastore.prepare(query);
+    Entity entity = results.asSingleEntity();
+    if (entity != null) {
+        response.sendRedirect("/index.html");
+        return;
+    }
+
+    String nickname = request.getParameter("nickname");
+    Entity userEntity = new Entity("User");
+    userEntity.setProperty("id", id);
+    userEntity.setProperty("nickname", nickname);
+    datastore.put(userEntity);
```

Comment:
Please check if this return an error

Reply:
I did some research on the exceptions possible with Datastore. The documentation does not indicate that the construction of a DatastoreService can create an error, so I believe it will, at worst, create an unusable connection. However, .put and .get can both throw exceptions on failed queries. I intend to add try/catches to each of these transactions, in addition to a new error page in a separate PR. Would that be okay?

Reply:
Ok, good research. Let us separately handle the errors/exceptions.